### ISSUE TITLE
Crashes when opening certain dialogs

### ISSUE THREAD

**Issue by da2x** [flathub/com.valvesoftware.Steam#265]
1. Go to Settings: Web browser
2. Change the URL of the home page and save

I briefly see a prompt to restart and then Steam crashes. This happens for every dialog that triggers this prompt.

1. Start any game running on Proton

There is a Steam-dialog opening saying something about FAQs and then Steam crashes the same way as it does for other Steam-dialogs.

Install dialogs work fine. Other dialogs work fine. These two types of dialogs seemto trigger a crash.

    GameAction [AppID 298610, ActionID 1] : LaunchApp changed task to ShowSteamPlayWarning with ""
    GameAction [AppID 298610, ActionID 1] : LaunchApp waiting for user response to ShowSteamPlayWarning ""
    Failed to open digital audio driver No mixer services exist.
    crash_20181129165022_1.dmp[320]: Uploading dump (out-of-process)
    /tmp/dumps/crash_20181129165022_1.dmp
    /home/user/.var/app/com.valvesoftware.Steam/.local/share/Steam/steam.sh: line 889:   126 Segmentation fault      (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
    crash_20181129165022_1.dmp[320]: Finished uploading minidump (out-of-process): success = yes
    crash_20181129165022_1.dmp[320]: response: CrashID=bp-8548e907-128c-47e6-b1a6-xxxxxxxxxxxxx
    crash_20181129165022_1.dmp[320]: file ''/tmp/dumps/crash_20181129165022_1.dmp'', upload yes: ''CrashID=bp-8548e907-128c-47e6-b1a6-xxxxxxxxxxxxx''

Fedora 29

**Comment by LeandroStanger** [comment]
Steam Beta?

**Comment by da2x** [comment]
Stable.

**Comment by nanonyme** [comment]
Hnm, need to take the time soon to investigate and document process in how to get traces from crashes in least painful way. Does this still repro

**Comment by nanonyme** [comment]
Can you take a screenshot about this FAQ bit?

**Comment by da2x** [comment]
Dialog that appears for a second after trying to launch any Steam Play games.
![steam-play-dialog](https://user-images.githubusercontent.com/1102886/49409664-d8716a80-f761-11e8-8560-b5d5328a0c38.png)

Dialog that appears after opening Settings and then just clicking OK:
![steam-settings-restart](https://user-images.githubusercontent.com/1102886/49409665-da3b2e00-f761-11e8-9293-92f3de3dd9b4.png)

There is a crash immediately following either of these dialogs. I can play Linux native titles just fine, however.

**Comment by nanonyme** [comment]
Does this still reproduce?

**Comment by da2x** [comment]
Yes. Ever time in Wayland sessions, and almost every time in X11.

**Comment by nanonyme** [comment]
Thanks. Was just checking since there's been a lot of Proton changes since.

**Comment by qcloutier** [comment]
I'm having the same issue (on Gentoo). When I try to enable Proton for all games in Settings > Steam Play, it crashes after briefly displaying the 'Restart Steam' dialog. 

`Installing breakpad exception handler for appid(steam)/version(1565656602)
AL lib: (EE) ALCplaybackOSS_open: Could not open /dev/dsp: No such file or directory
Failed to open digital audio driver Couldn't open OAL device.
crash_20190814205416_1.dmp[444]: Uploading dump (out-of-process)
/tmp/dumps/crash_20190814205416_1.dmp
/home/user/.local/share/Steam/steam.sh: line 720:   307 Segmentation fault      $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
crash_20190814205416_1.dmp[444]: Finished uploading minidump (out-of-process): success = yes
crash_20190814205416_1.dmp[444]: response: CrashID=bp-0e2a67a7-7d59-41d4-9995-xxxxxxxxxxxxx
crash_20190814205416_1.dmp[444]: file ''/tmp/dumps/crash_20190814205416_1.dmp'', upload yes: ''CrashID=bp-0e2a67a7-7d59-41d4-9995-xxxxxxxxxxxxx''`

**Comment by qcloutier** [comment]
After thinking about this for a bit (and reading the error a little more closely), I realized that Steam is using OSS (for some reason) to play a sound when any of these dialogs appear. `modprobe snd-pcm-oss` loads the needed module and fixes the crash.

I did not have this issue previously when Steam was installed on my system from the Gentoo Steam overlay. Steam played the sounds just fine despite /dev/dsp not existing. Furthermore, on my Arch machine running Steam from its repos, Steam doesn't play the sound at all.
